In [2]:
import pandas as pd

In [3]:
titles = pd.read_csv('C:/Users/comp/Python Coding/LU AI-ML/Numpy, Pandas etc/Day - 16, 17/Assingment/Dataset for Pandas Assinment/titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

In [4]:
cast = pd.read_csv('C:/Users/comp/Python Coding/LU AI-ML/Numpy, Pandas etc/Day - 16, 17/Assingment/Dataset for Pandas Assinment/cast.csv')
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4  31.0
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

In [5]:
rel_date = pd.read_csv('C:/Users/comp/Python Coding/LU AI-ML/Numpy, Pandas etc/Day - 16, 17/Assingment/Dataset for Pandas Assinment/release_dates.csv')
rel_date.head()

title  year      country        date
0   #73, Shaanthi Nivaasa  2007        India  2007-06-15
1                   #BKKY  2016     Cambodia  2017-10-12
2                 #Beings  2015      Romania  2015-01-29
3               #Captured  2017          USA  2017-09-05
4  #Ewankosau saranghaeyo  2015  Philippines  2015-01-21

### How many movies are listed in the titles dataframe?

In [129]:
len(titles['title'])

244914

There are a total movie titles of 244914 listed

### What are the earliest two films listed in the titles dataframe?

In [17]:
df1 = titles.sort_values(by='year', ascending=True)

df1.head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

The above are the earliset two films listed in the titles dataframe

### How many movies have the title "Hamlet"?

In [135]:
titles[titles['title'] =='Hamlet'].title.count()
print(a.title.count(),' movies have their title listed as "Hamlet" ')

20  movies have their title listed as "Hamlet" 


### How many movies are titled "North by Northwest"?

In [134]:
print('The number of movies with title North by Northwest is: ', titles[titles['title'] == 'North by Northwest'].title.count())

The number of movies with title North by Northwest is:  1


### When was the first movie titled "Hamlet" made?

In [178]:
b = titles[titles['title'] == "Hamlet"].groupby('year').min()
print("First movie titled 'Hamlet' was made in year : \n\n", b.head(1))

First movie titled 'Hamlet' was made in year : 

        title
year        
1910  Hamlet


### List all of the "Treasure Island" movies from earliest to most recent.

In [83]:
titles[titles['title'] == 'Treasure Island'].groupby('year').min()

title
year                 
1918  Treasure Island
1920  Treasure Island
1934  Treasure Island
1950  Treasure Island
1972  Treasure Island
1973  Treasure Island
1985  Treasure Island
1999  Treasure Island

The above are all the movies titled "Treasure Island" from earliest to most recent

### How many movies were made in the year 1950?

In [9]:
print("The number of movies made in year 1950 are :", titles[titles['year'] == 1950].title.count())

The number of movies made in year 1950 are : 1113


### How many movies were made in the year 1960?

In [10]:
print('The number of movies made in year 1960 are :', titles[titles['year'] == 1960].title.count())

The number of movies made in year 1960 are : 1549


### How many movies were made from 1950 through 1959?

In [11]:
print('The number of movies made between 1950 & 1959 are: ', titles[(titles['year'] > 1949) & (titles['year'] < 1960)].title.count())

The number of movies made between 1950 & 1959 are:  12934


### In what years has a movie titled "Batman" been released?

In [32]:
print('The movie title "Batman" has been released in the year: \n\n', titles[titles['title'] == 'Batman'].groupby('year').sum())

The movie title "Batman" has been released in the year: 

        title
year        
1943  Batman
1989  Batman


### How many roles were there in the movie "Inception"?

In [17]:
print('The number of roles in the movie "Inception" are :', cast[cast['title'] == 'Inception'].title.count())

The number of roles in the movie "Inception" are : 78


### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [23]:
print('Number of roles in the movie "Inception" are NOT ranked by a "n" value are :', cast[(cast['title'] == "Inception") & (cast['n'].isna() == True)].title.count())

Number of roles in the movie "Inception" are NOT ranked by a "n" value are : 27


### But how many roles in the movie "Inception" did receive an "n" value?

In [25]:
print('The number of roles in movie "Inception" to receive "n" value are :', cast[(cast['title'] == 'Inception') & (cast['n'].isna() == False)].title.count())

The number of roles in movie "Inception" to receive "n" value are : 51


### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [51]:
print('Displaying the names of cast of "North by Northwest" in their correct "n" value order :')
cast[(cast['title'] == "North by Northwest") & (cast['n'].isna() == False)].sort_values(by='n')

Displaying the names of cast of "North by Northwest" in their correct "n" value order :


title  year                  name     type  \
871426   North by Northwest  1959            Cary Grant    actor   
3518333  North by Northwest  1959       Eva Marie Saint  actress   
1457165  North by Northwest  1959           James Mason    actor   
3157232  North by Northwest  1959   Jessie Royce Landis  actress   
356707   North by Northwest  1959        Leo G. Carroll    actor   
3048272  North by Northwest  1959  Josephine Hutchinson  actress   
1695173  North by Northwest  1959           Philip Ober    actor   
1275337  North by Northwest  1959         Martin Landau    actor   
2447081  North by Northwest  1959         Adam Williams    actor   
1811387  North by Northwest  1959          Edward Platt    actor   
665768   North by Northwest  1959     Robert Ellenstein    actor   
2295714  North by Northwest  1959          Les Tremayne    actor   
464736   North by Northwest  1959       Philip Coolidge    actor   
1508344  North by Northwest  1959         Patrick McVey    actor   
206471   North by Northwest  1959          Edward Binns    actor   
1384780  North by Northwest  1959             Ken Lynch    actor   

                           character     n  
871426            Roger O. Thornhill   1.0  
3518333                  Eve Kendall   2.0  
1457165              Phillip Vandamm   3.0  
3157232              Clara Thornhill   4.0  
356707                 The Professor   5.0  
3048272                Mrs. Townsend   6.0  
1695173              Lester Townsend   7.0  
1275337                      Leonard   8.0  
2447081                     Valerian   9.0  
1811387              Victor Larrabee  10.0  
665768                         Licht  11.0  
2295714                   Auctioneer  12.0  
464736                     Dr. Cross  13.0  
1508344               Sergeant Flamm  14.0  
206471                Captain Junket  15.0  
1384780  Charley - Chicago Policeman  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [60]:
print('Below listed are the cast of the movie "Sleuth" from year 1972, in "n" order :')
cast[(cast['title'] == 'Sleuth') & (cast['year'] == 1972)].sort_values(by='n')

Below listed are the cast of the movie "Sleuth" from year 1972, in "n" order :


title  year                name     type  \
1705394  Sleuth  1972    Laurence Olivier    actor   
326773   Sleuth  1972       Michael Caine    actor   
373876   Sleuth  1972      Alec Cawthorne    actor   
1465483  Sleuth  1972  John (II) Matthews    actor   
2724553  Sleuth  1972  Eve (III) Channing  actress   
1448690  Sleuth  1972        Teddy Martin    actor   

                          character    n  
1705394                 Andrew Wyke  1.0  
326773                  Milo Tindle  2.0  
373876            Inspector Doppler  3.0  
1465483  Detective Sergeant Tarrant  4.0  
2724553             Marguerite Wyke  5.0  
1448690      Police Constable Higgs  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [61]:
print('Below displayed is the entire cast of the 2007 "Sleuth" movie in the "n" order :')
cast[(cast['title'] == "Sleuth") & (cast['year'] == 2007)].sort_values(by='n')

Below displayed is the entire cast of the 2007 "Sleuth" movie in the "n" order :


title  year                 name     type          character    n
326774   Sleuth  2007        Michael Caine    actor             Andrew  1.0
1293634  Sleuth  2007             Jude Law    actor               Milo  2.0
1805907  Sleuth  2007        Harold Pinter    actor        Man on T.V.  3.0
260242   Sleuth  2007      Kenneth Branagh    actor  Other Man on T.V.  NaN
373877   Sleuth  2007  Alec (II) Cawthorne    actor  Inspector Doppler  NaN
2724552  Sleuth  2007    Eve (II) Channing  actress    Marguerite Wyke  NaN
3370424  Sleuth  2007    Carmel O'Sullivan  actress             Maggie  NaN

### How many roles were credited in the silent 1921 version of Hamlet?

In [66]:
print('Number of roles credited in the silent 1921 version of "Hamlet" are :', cast[(cast['title'] == "Hamlet") & (cast['year'] == 1921)].name.count())

Number of roles credited in the silent 1921 version of "Hamlet" are : 9


### How many roles were credited in Branagh’s 1996 Hamlet?

In [69]:
print('Roles credited in Branagh`s 1996 Hamlet are :', cast[(cast['title'] == "Hamlet") & (cast['year'] == 1996)].name.count())

Roles credited in Branagh`s 1996 Hamlet are : 54


### How many "Hamlet" roles have been listed in all film credits through history?

In [72]:
print('THe number of "Hamlet" roles listed in all film credits throughout history are :', cast[cast['title'] == "Hamlet"].name.count())

THe number of "Hamlet" roles listed in all film credits throughout history are : 342


### How many people have played an "Ophelia"?

In [74]:
print('Number of people that have played "Ophelia" are :', cast[cast['character'] == "Ophelia"].character.count())

Number of people that have played "Ophelia" are : 117


### How many people have played a role called "The Dude"?

In [75]:
print('The number of people that have played "The Dude" are :', cast[cast['character'] == "The Dude"].character.count())

The number of people that have played "The Dude" are : 19


### How many people have played a role called "The Stranger"?

In [76]:
print('The number of people that have played the role called "The Stranger" are :', cast[cast['character'] == "The Stranger"].character.count())

The number of people that have played the role called "The Stranger" are : 212


### How many roles has Sidney Poitier played throughout his career?

In [79]:
print('The number of roles played by "Sidney Poitier" throughout his career are :', cast[cast['name'] == "Sidney Poitier"].title.count())

The number of roles played by "Sidney Poitier" throughout his career are : 43


### How many roles has Judi Dench played?

In [80]:
print('The number of roles played by "Judi Dench" are :', cast[cast['name'] == "Judi Dench"].title.count())

The number of roles played by "Judi Dench" are : 55


### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [89]:
print('Supporting Roles played by "Cary Grant" in 1940s are listed below :')
cast[(cast['name'] == "Cary Grant") & (cast['n'] == 2) & ((cast['year'] >= 1940) & (cast['year'] < 1950))].sort_values(by='year')

Supporting Roles played by "Cary Grant" in 1940s are listed below :


title  year        name   type    character    n
871423  My Favorite Wife  1940  Cary Grant  actor   Nick Arden  2.0
871433    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [93]:
print('The list of roles played by Cary Grant in the 1940s are :')
cast[(cast['name'] == "Cary Grant") & ((cast['year'] >= 1940) & (cast['year'] < 1950))].sort_values(by='year')

The list of roles played by Cary Grant in the 1940s are :


title  year        name   type  \
871450                The Philadelphia Story  1940  Cary Grant  actor   
871448               The Howards of Virginia  1940  Cary Grant  actor   
871407                       His Girl Friday  1940  Cary Grant  actor   
871423                      My Favorite Wife  1940  Cary Grant  actor   
871438                             Suspicion  1941  Cary Grant  actor   
871433                        Penny Serenade  1941  Cary Grant  actor   
871452                  The Talk of the Town  1942  Cary Grant  actor   
871429                 Once Upon a Honeymoon  1942  Cary Grant  actor   
871398                     Destination Tokyo  1943  Cary Grant  actor   
871422                             Mr. Lucky  1943  Cary Grant  actor   
871390                  Arsenic and Old Lace  1944  Cary Grant  actor   
871430                      Once Upon a Time  1944  Cary Grant  actor   
871425             None But the Lonely Heart  1944  Cary Grant  actor   
871405               George White's Scandals  1945  Cary Grant  actor   
871424                         Night and Day  1946  Cary Grant  actor   
871427                             Notorious  1946  Cary Grant  actor   
871466                  Without Reservations  1946  Cary Grant  actor   
871444      The Bachelor and the Bobby-Soxer  1947  Cary Grant  actor   
871445                     The Bishop's Wife  1947  Cary Grant  actor   
871421  Mr. Blandings Builds His Dream House  1948  Cary Grant  actor   
871402          Every Girl Should Be Married  1948  Cary Grant  actor   
871411                I Was a Male War Bride  1949  Cary Grant  actor   

                                   character    n  
871450                    C. K. Dexter Haven  1.0  
871448                           Matt Howard  1.0  
871407                          Walter Burns  1.0  
871423                            Nick Arden  2.0  
871438                      Johnnie Aysgarth  1.0  
871433                           Roger Adams  2.0  
871452                          Leopold Dilg  1.0  
871429                 Patrick 'Pat' O'Toole  1.0  
871398                         Capt. Cassidy  1.0  
871422  Joe Adams -posing as Joe Bascopolous  1.0  
871390                     Mortimer Brewster  1.0  
871430                           Jerry Flynn  1.0  
871425                            Ernie Mott  1.0  
871405                            Cary Grant  NaN  
871424                           Cole Porter  1.0  
871427                                Devlin  1.0  
871466                            Cary Grant  NaN  
871444                           Dick Nugent  1.0  
871445                                Dudley  1.0  
871421                         Jim Blandings  1.0  
871402                     Dr. Madison Brown  1.0  
871411                 Captain Henri Rochard  1.0

### How many roles were available for actors in the 1950s?

In [98]:
print('Number of roles available for actors in the 1950s are :', cast[(cast['type'] == 'actor') & ((cast['year'] >= 1950) & (cast['year'] < 1960))].type.count())

Number of roles available for actors in the 1950s are : 157738


### How many roles were available for actresses in the 1950s?

In [100]:
print('The number of roles available for actresses in the 1950s are :', cast[(cast['type'] == "actress") & ((cast['year'] >= 1950) & (cast['year'] < 1960))].type.count())

The number of roles available for actresses in the 1950s are : 57733


### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [102]:
print('Number of leading roles available from the beginning of film history upto 1980 is :', cast[(cast['n'] == 1) & (cast['year'] < 1981)].n.count())

Number of leading roles available from the beginning of film history upto 1980 is : 65140


### How many non-leading roles were available through from the beginning of film history through 1980?

In [119]:
print('Number of non-leading roles available through from the beginning of film history upto 1980 is :', len(cast[(cast['n'] != 1) & (cast['year'] < 1981)].n))

Number of non-leading roles available through from the beginning of film history upto 1980 is : 1117667


### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [127]:
print("Number of roles in 1980 that were minor enough and didn't warrant a rank :", len(cast[(cast['n'].isna() == True) & (cast['year'] == 1980)]))

Number of roles in 1980 that were minor enough and didn't warrant a rank : 4466
